miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -284,25 +303,83 @@ internal class MiniAppDownloader(
         }
     }
 
+    @SuppressWarnings("LongMethod", "NestedBlockDepth", "ComplexMethod", "ThrowsCount")
+    internal suspend fun downloadMiniApp(
+        appId: String,
+        versionId: String,
+    ): String {
+        val manifest = fetchManifest(appId, versionId)
+        val baseSavePath = storage.getMiniAppVersionPath(appId, versionId)
+        when {
+            doesManifestFileExist(manifest.first) -> {
+                for (file in manifest.first.files) {
+                    try {
+                        checkSignatureValidation(
+                            file = file,
+                            versionId = versionId,
+                            manifest = manifest,
+                            appId = appId
+                        )
+                        storage.saveFile(
+                            file,
+                            baseSavePath,
+                            apiClient.downloadFile(file).byteStream()
+                        )
+                    } catch (error: MiniAppTooManyRequestsError) {
+                        removeMiniApp(appId, versionId, TOO_MANY_REQUEST_ERR_LOG)
```

Comment:
This is added for the QPS right? This will remove all other versions. We don't need this I guess

Reply:
i don't know why this is here but it's specific for too many request and if it happened it will remove that miniapp, not the other versions. I think we can remove it.

Reply:
updated.